packages/core/src/utils/google-genai/types.ts

```diff
@@ -0,0 +1,193 @@
+  /** Total token count for prompt, response candidates, and tool-use prompts (if present). */
+  totalTokenCount?: number;
+};
+
+/**
+ * Google GenAI Candidate
+ * @see https://ai.google.dev/api/rest/v1/Candidate
+ * https://github.com/googleapis/js-genai/blob/v1.19.0/src/types.ts#L2237
+ */
+export type Candidate = {
+  [key: string]: unknown;
+  /**
+   * Contains the multi-part content of the response.
+   */
+  content?: Content;
+  /**
+   * The reason why the model stopped generating tokens.
+   * If empty, the model has not stopped generating the tokens.
+   */
+  finishReason?: string;
+  /**
+   * Number of tokens for this candidate.
+   */
+  tokenCount?: number;
+  /**
+   * The index of the candidate.
+   */
+  index?: number;
+};
+
+/**
+ * Google GenAI Generate Content Response
+ * @see https://ai.google.dev/api/rest/v1/GenerateContentResponse
+ */
+type GenerateContentResponse = {
+  [key: string]: unknown;
+  /** Response variations returned by the model. */
+  candidates?: Candidate[];
+  /** Timestamp when the request is made to the server. */
+  automaticFunctionCallingHistory?: Content[];
+  /** Output only. The model version used to generate the response. */
+  modelVersion?: string;
+  /** Output only. Content filter results for a prompt sent in the request. Note: Sent only in the first stream chunk. Only happens when no candidates were generated due to content violations. */
+  promptFeedback?: Record<string, unknown>;
+  /** Output only. response_id is used to identify each response. It is the encoding of the event_id. */
+  responseId?: string;
+  /** Usage metadata about the response(s). */
+  usageMetadata?: GenerateContentResponseUsageMetadata;
+};
+
+/**
+ * Basic interface for Google GenAI client with only the instrumented methods
+ * This provides type safety while being generic enough to work with different client implementations
+ */
+export interface GoogleGenAIClient {
+  models: {
+    generateContent: (...args: unknown[]) => Promise<GenerateContentResponse>;
+    // https://googleapis.github.io/js-genai/release_docs/classes/models.Models.html#generatecontentstream
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    generateContentStream: (...args: unknown[]) => Promise<AsyncGenerator<GenerateContentResponse, any, unknown>>;
+  };
+  chats: {
+    create: (...args: unknown[]) => GoogleGenAIChat;
+  };
+}
+
+/**
+ * Google GenAI Chat interface for chat instances created via chats.create()
+ */
+export interface GoogleGenAIChat {
+  sendMessage: (...args: unknown[]) => Promise<GenerateContentResponse>;
+  // https://googleapis.github.io/js-genai/release_docs/classes/chats.Chat.html#sendmessagestream
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  sendMessageStream: (...args: unknown[]) => Promise<AsyncGenerator<GenerateContentResponse, any, unknown>>;
+}
+
+/**
+ * Google GenAI Integration interface for type safety
+ */
+export interface GoogleGenAIIntegration {
```

Comment:
do we actually need this, when we do not lookup the integration on the client anymore? :)